drivers/hmis/app/models/hmis/ce/referral_enroller.rb

```diff
@@ -26,47 +26,81 @@ def create_enrollment(message)
-
-      # Raise an exception if there are any entry date validation errors, which may mean that the client is already enrolled at the project.
-      # Future improvement would be to return this as a validation error instead of raising
-      entry_date_errors = Hmis::Hud::Validators::EnrollmentValidator.validate_entry_date(enrollment)
-      entry_date_errors.reject!(&:warning?)
-      error_out(entry_date_errors.map(&:full_message).join(', ')) unless entry_date_errors.empty?
+      # clients_to_enroll maps Hmis::Hud::Client to their integer Relationship to HoH. It includes the referred client (HoH)
+      clients_to_enroll = get_clients_to_enroll(referral.source_enrollment)
 
+      # Generate a new household ID for the target enrollment household
+      new_household_id = Hmis::Hud::Base.generate_uuid
+      hud_user = Hmis::Hud::User.from_user(message.user)
+      entry_date = Date.current
       # TODO(#7537) - prevent conflicting unit occupancy
       unit = referral.opportunity.unit
-      enrollment.assign_unit(unit: unit, start_date: Date.current, user: message.user) if unit.is_a? Hmis::Unit
 
-      enrollment.save_new_enrollment! # Saves as WIP or non-WIP, depending on auto-enter rules in the project
-      referral.update!(target_enrollment: enrollment)
+      validation_errors = []
+
+      # Create new enrollments for all household members
+      enrollments = clients_to_enroll.map do |client, relationship_to_hoh|
+        enrollment = Hmis::Hud::Enrollment.new(
+          client: client,
+          project: project,
+          entry_date: entry_date,
+          user: hud_user,
+          household_id: new_household_id,
+          relationship_to_hoh: relationship_to_hoh,
+          enrollment_coc: coc_code,
+        )
+
+        raise 'referral generated invalid enrollment' unless enrollment.valid?
+
+        # Collect entry date validations, which may indicate that the client is already enrolled at the project.
+        entry_date_errors = Hmis::Hud::Validators::EnrollmentValidator.validate_entry_date(enrollment)
+        entry_date_errors.reject!(&:warning?)
+        validation_errors.concat(entry_date_errors)
+
+        enrollment
+      end
+
+      # Raise an exception if there are any entry date validation errors.
+      # Future improvement would be to return this as a validation error instead of raising
+      error_out(validation_errors.map(&:full_message).join(', ')) unless validation_errors.empty?
+
+      # Assign the HoH's (referred client's) enrollment to the unit and save.
+      # Do this before saving the other hh member enrollments, so that assign_unit validations pass.
+      hoh_target_enrollment = enrollments.find { |e| e.client == referral.client }
+      enrollments.delete(hoh_target_enrollment) # Remove from list so it's not saved again below
+      hoh_target_enrollment.assign_unit(unit: unit, start_date: entry_date, user: message.user)
```

Comment:
probably want this here for consistency?

```suggestion
      hoh_target_enrollment.assign_unit(unit: unit, start_date: entry_date, user: message.user, active_referral: referral)
```

Reply:
Thanks! Opened https://github.com/greenriver/hmis-warehouse/pull/5931 as a follow-up